pages/en/learn/test-runner/using-test-runner.md

```diff
@@ -0,0 +1,312 @@
+
+There is no specific configuration _required_ except enabling the feature via [`--experimental-test-snapshots`](). But to demonstrate the optional configuration, you would probably add something like the following to one of your existing test config files.
+
+<details>
+<summary>`test/setup.ui.mjs`</summary>
+
+By default, node generates a filename that is incompatible with syntax highlighting detection: `.js.snapshot`. The generated file is actually a CJS file, so a more appropriate file name would end with `.snapshot.cjs` (or more succinctly `.snap.cjs` as below); this will also handle better in ESM projects.
+
+```js
+import { basename, dirname, extname, join } from 'node:path';
+import { snapshot } from 'node:test';
+
+snapshot.setResolveSnapshotPath(generateSnapshotPath);
+/**
+ * @param {string} testFilePath '/tmp/foo.test.js'
+ * @returns {string} '/tmp/foo.test.snap.cjs'
+ */
+function generateSnapshotPath(testFilePath) {
+  const ext = extname(testFilePath);
+  const filename = basename(testFilePath, ext);
+  const base = dirname(testFilePath);
+
+  return join(base, `${filename}.snap.cjs`);
+}
+```
+
+</details>
+
+The example below demonstrates snapshot testing with [testing library](https://testing-library.com/) for UI components; note the two different ways of accessing `assert.snapshot`):
+
+```ts
+import { describe, it } from 'node:test';
+
+import { prettyDOM } from '@testing-library/dom';
+import { render } from '@testing-library/react'; // Any framework (ex svelte)
+
+import { SomeComponent } from './SomeComponent.jsx';
+
+
+describe('<SomeComponent>', () => {
+  // For people preferring "fat-arrow" syntax, the following is probably better for consistency
+  it('should render defaults when no props are provided', (t) => {
+    const component = render(<SomeComponent />).container.firstChild;
+
+    t.assert.snapshot(prettyDOM(component));
+  });
+
+  it('should consume `foo` when provided', function() {
+    const component = render(<SomeComponent foo="bar" />).container.firstChild;
+
+    this.assert.snapshot(prettyDOM(component));
+    // `this` works only when `function` is used (not "fat arrow").
+  });
+});
+```
+
+> ⚠️ `assert.snapshot` comes from the test's context (`t` or `this`), **not** `node:assert`. This is necessary because the test context has access to scope that is impossible for `node:assert` (you would have to manually provide it every time `assert.snapshot` is used, like `snapshot(this, value)`, which would be rather tedious).
+
+## Unit tests
+
+Unit tests are the simplest tests and generally require relatively nothing special. The vast majority of your tests will likely be unit tests, so it is important to keep this setup minimal because a small decrease to setup performance will magnify and cascade.
```

Comment:
I wonder if the examples should lead with this - it's mentioned as the most common use case.

Reply:
It could do. @AugustinMauroy mentioned it may be good to have a "you _should_" parts. I think that may be a good time to move this elsewhere.

Reply:
this paragraph could be combined in `introduction.md`. what do you think ?

Reply:
I dont feel so strongly that my one comment should be blocking merge